### PR TITLE
Gitlab Integration Support

### PR DESCRIPTION
**Pull request by Shasidhar**
Adding gitlab_ci yml file which will extend the porject support to Gitlab CI-CD
The idea is: 
1. On every push to all branches we run unit-tests and integration tests
2. On push to master we deploy the actual job along with running all tests

---
This change is [Reviewable](https://reviewable.io/reviews/databrickslabs/cicd-templates/48)

### CODE REVIEW

File: {{cookiecutter.project_slug}}/.gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+integration-testing:
+  stage: integration-testing
+  image: python:3.7-stretch
+  script:
+    - echo "Install dependencies"
+    - pip install -r unit-requirements.txt
+    - pip install -U tools/dbx-0.7.0-py3-none-any.whl
+    - pip install -e .
+    - echo "Preparing profile"
+    - echo "[DEFAULT]" >> ~/.databrickscfg
```
Comment (renardeinside) [inline {{cookiecutter.project_slug}}/.gitlab-ci.yml:28]: Profile name should be dynamic, based on {{cookiecutter.profile}} parameter. Please check the Azure Pipelines file for reference.

File: {{cookiecutter.project_slug}}/.gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  only:
+    refs:
+      - master
+  script:
+    - echo "Install dependencies"
+    - pip install -r unit-requirements.txt
+    - pip install -U tools/dbx-0.7.0-py3-none-any.whl
+    - pip install -e .
+    - echo "Preparing profile"
+    - echo "[DEFAULT]" >> ~/.databrickscfg
```
Comment (renardeinside) [inline {{cookiecutter.project_slug}}/.gitlab-ci.yml:48]: Profile name should be dynamic, based on {{cookiecutter.profile}} parameter. Please check the Azure Pipelines file for reference.

**Review by renardeinside — CHANGES_REQUESTED** [review]
Great job @Shasidhar ! Please take a look at some small adjustments about profile naming.

File: {{cookiecutter.project_slug}}/.gitlab-ci.yml (hunk 2 above)
Comment (Shasidhar) [reply {{cookiecutter.project_slug}}/.gitlab-ci.yml]: Fixed

File: {{cookiecutter.project_slug}}/.gitlab-ci.yml (hunk 1 above)
Comment (Shasidhar) [reply {{cookiecutter.project_slug}}/.gitlab-ci.yml]: Fixed

**Comment by renardeinside** [comment]
@Shasidhar thanks a lot again!

**Pull request merged**